deepchem/data/data_loader.py

```diff
@@ -1613,3 +1616,23 @@ def _featurize_shard(  # type: ignore[override]
             ids.append(entry_id)
         X = np.concatenate(features, axis=0)
         return X, np.array(labels), np.array(weights), np.array(ids)
+
+class DFTYamlLoader(DataLoader):
+    def __init__(self,
+                 featurizer: Featurizer):
+        tasks = None
+    def create_dataset(self, input_files: str, featurizer):
+        entries = self._get_shards(input_files)
+        y = np.array([self._featurize_shard(shard) for shard in entries] )
```

Comment:
This is not correct; a `DataLoader` should return a DeepChem `Dataset` object and not a numpy array. Check the other data loaders to check their return types